lib/base.js

```diff
@@ -244,36 +244,30 @@ class ConnectionPool extends EventEmitter {
     this._poolCreate().then((connection) => {
       debug('pool(%d): connected', IDS.get(this))
 
-      return this._poolDestroy(connection).then(() => {
-        if (!this._connecting) {
-          debug('pool(%d): not connecting, exiting silently (was close called before connection established?)', IDS.get(this))
-          return
-        }
+      this._poolDestroy(connection)
+      if (!this._connecting) {
+        debug('pool(%d): not connecting, exiting silently (was close called before connection established?)', IDS.get(this))
+        return
+      }
 
-        // prepare pool
-        this.pool = gp.createPool({
+      // prepare pool
+      this.pool = new tarn.Pool(
+        Object.assign({
           create: this._poolCreate.bind(this),
           validate: this._poolValidate.bind(this),
-          destroy: this._poolDestroy.bind(this)
-        }, Object.assign({
+          destroy: this._poolDestroy.bind(this),
           max: 10,
           min: 0,
-          evictionRunIntervalMillis: 1000,
+          acquireTimeoutMillis: this.config.connectionTimeout || this.config.timeout || 15000,
           idleTimeoutMillis: 30000,
-          testOnBorrow: true
-        }, this.config.pool))
-
-        this.pool.on('factoryCreateError', this.emit.bind(this, 'error'))
-        this.pool.on('factoryDestroyError', this.emit.bind(this, 'error'))
+          propagateCreateError: true
+        }, this.config.pool)
+      )
 
-        this._connecting = false
-        this._connected = true
-
-        callback(null)
-      })
-    }).catch(err => {
```

Comment:
Removing the catch here makes the initial probe poolCreate an uncaught rejection.

Reply:
I've added this back